packages/reports/src/reports/blocks/types/adapters/Selector.js

```diff
@@ -0,0 +1,37 @@
+import {SELECTOR_TYPES} from "../../../../utils/consts/cms";
+import valueInOptions from "../../../../utils/selectors/valueInOptions";
+
+const validateOptionsArray = options => {
+  if (!Array.isArray(options)) return [];
+  return options.map(d => typeof d === "string" ? {id: d, label: d} : d);
+};
+
+const maybeFixForMulti = (defaultValue, type) => 
+  type === SELECTOR_TYPES.MULTI && !Array.isArray(defaultValue)
+    ? [defaultValue]
+    : defaultValue;
+
+/** @type {import(".").BlockPreviewAdapterFunction} */
+export default (vars, id) => {
```

Comment:
I am realizing that the longer I spend away from these functions, the more I forget what they do. In this case I realize the Selector function is an adapter - where is the proper place to document exactly what's happening here? I'd like to see a function description here - and perhaps a similar one in the other adapters. I'm a big fan of putting examples in the comment doc: "this function takes variables calculated by X thing and exports Y config, and here's an example of what that might look like"

I know for example this note to myself saved me a bunch of times: https://github.com/Datawheel/canon/blob/master/packages/cms/src/utils/varSwapRecursive.js#L42-L46

Wondering your thoughts on how to standardize these kinds of function recipe/descriptions?

Reply:
There is one! See the index.js file in this same directory. That `BlockPreviewAdapterFunction` typedef has a short description of what this function does, though it could be improved.

Basically, this is a function that is called to transform the output of a block's evaluated logic to the props of the BlockType's renderer React component. This is an optional thing where one might want to write logic for programming fallbacks or turning the output variables into something that is only needed for rendering.